torch/distributed/checkpoint/filesystem.py

```diff
@@ -479,6 +479,9 @@ def exists(self, path: Union[str, os.PathLike]) -> bool: ...
     @abstractmethod
     def rm_file(self, path: Union[str, os.PathLike]) -> None: ...
 
+    @abstractmethod
+    def ls(self, path: Union[str, os.PathLike]) -> list[str]: ...
```

Comment:
By adding an additional abstract method to this abstract base class, this PR broke all users who subclassed `FileSystemBase` since such subclasses need to implement all abstract methods. Could you find a fix or revert this PR? Thanks